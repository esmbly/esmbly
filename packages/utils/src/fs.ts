import fs from 'fs';
import util from 'util';

export default {
  readdir: util.promisify(fs.readdir),
  readFile: util.promisify(fs.readFile),
  stat: util.promisify(fs.stat),
  writeFile: util.promisify(fs.writeFile),
};
