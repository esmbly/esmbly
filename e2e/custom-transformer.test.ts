import fs from 'fs';
import util from 'util';
import path from 'path';
import { testRunner } from './helpers/testRunner';

const readFile = util.promisify(fs.readFile);
const example = path.resolve(path.join('examples', 'custom-transformer'));
const dist = path.join(example, 'dist');

describe('E2E: custom-transformer', () => {
  beforeAll(async () => {
    await testRunner(example);
  });

  it('outputs a TypeScript file where the foo function has been renamed to bar', async () => {
    const result = await readFile(path.join(dist, 'bar.ts'));
    expect(result).toMatchSnapshot();
  });
});
