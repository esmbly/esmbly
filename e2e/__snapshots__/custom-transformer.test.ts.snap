// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`E2E: custom-transformer outputs a TypeScript file where foo-functions has been renamed to bar 1`] = `
Object {
  "data": Array [
    101,
    120,
    112,
    111,
    114,
    116,
    32,
    102,
    117,
    110,
    99,
    116,
    105,
    111,
    110,
    32,
    98,
    97,
    114,
    40,
    41,
    58,
    32,
    115,
    116,
    114,
    105,
    110,
    103,
    32,
    123,
    10,
    32,
    32,
    114,
    101,
    116,
    117,
    114,
    110,
    32,
    39,
    98,
    97,
    114,
    39,
    59,
    10,
    125,
    10,
  ],
  "type": "Buffer",
}
`;
